log-window.js

```diff
@@ -26,261 +26,87 @@ class EmeLogWindow {
-        data.textContent += '\nAssociated value: ' + prettyPrint(log.value);
-      }
-    }
-
-    const textBasedLog =
-        formattedTimestamp + '\n\n' +
-        instanceId.textContent + '\n' +
-        data.textContent + '\n\n\n\n';
-
-    this.textLogs_ += textBasedLog;
   }
 }
 
 EmeLogWindow.instance = new EmeLogWindow();
-window.EmeLogWindow = EmeLogWindow;
-
-
-/**
- * @param {number} byte
- * @return {string}
- */
-function byteToHex(byte) {
-  return '0x' + byte.toString(16).padStart(2, '0');
-}
-
-/**
- * @param {*} obj
- * @param {string} indentation
- * @return {string}
- */
-function prettyPrint(obj, indentation = '') {
-  if (obj == null) {
-    return obj;
-  }
-
-  // If it's a named type, unpack it and attach the name.
-  if (obj.__type__) {
-    let format = obj.__type__ + ' instance';
-
-    // This has fields like an object.
-    if (obj.__fields__) {
-      format += ' ' + prettyPrint(obj.__fields__, indentation);
-    }
-
-    // This has a data array like an ArrayBufferView.
-    // TODO: Handle formatting for 16-bit and 32-bit values?
-    if (obj.__data__) {
-      const data = obj.__data__.slice();  // Make a copy
-      if (data.length == 0) {
-        format += '[]';
-      } else {
-        format += ' ' + '[\n';
-        while (data.length) {
-          const row = data.splice(0, 16);
-          format += indentation + '  ';
-          format += row.map(byteToHex).join(', ');
-          format += ',\n';
-        }
-        format += indentation + ']';
-      }
-    }
-    return format;
-  }
-
-  if (Array.isArray(obj)) {
-    // More compact representations for empty or 1-element arrays.
-    if (obj.length == 0) {
-      return '[]';
-    }
-    if (obj.length == 1) {
-      return `[${prettyPrint(obj[0], indentation)}]`;
-    }
-
-    let insides = '';
-    for (const entry of obj) {
-      insides += indentation + '  ';
-      insides += prettyPrint(entry, indentation + '  ') + ',\n';
-    }
-    return `[\n${insides}${indentation}]`;
-  }
-
-  if (obj.constructor == Object) {
-    const keys = Object.keys(obj);
-
-    // More compact representations for empty or 1-element objects.
-    if (keys.length == 0) {
-      return '{}';
-    }
-    if (keys.length == 1) {
-      return `{${keys[0]}: ${prettyPrint(obj[keys[0]], indentation)}}`;
-    }
-
-    let insides = '';
-    for (const key of keys) {
-      insides += indentation + '  ' + key + ': ';
-      insides += prettyPrint(obj[key], indentation + '  ') + ',\n';
-    }
-    return `{\n${insides}${indentation}}`;
-  }
-
-  if (typeof obj == 'string') {
-    return `"${obj}"`;
-  }
-
-  return obj.toString();
-}
 
 // NOTE: These APIs are not defined in our test environment, but should always
 // be present when this is run as a Chrome extension.
 if (chrome.runtime !== undefined) {
   /**
-   * Listens for messages from the content script to append a log item to the
-   * current frame and log file.
+   * When a window is closed, close the log window.
```

Comment:
This makes it sound like we close the log window when any other window is closed.  It seems instead that we close the log window _handle_ if the _actual_ log window was already closed.  Other closed window notifications will be ignored.

So the code all looks correct, but this comment seems a little misleading.

Reply:
Updated the description to "When a window is closed, clear the handle and logs if it was the log window."